stdlib/http/src/main/java/org/ballerinalang/net/http/websocket/client/WebSocketClientHandshakeListener.java

```diff
@@ -42,53 +39,52 @@
 public class WebSocketClientHandshakeListener implements ClientHandshakeListener {
 
     private final WebSocketService wsService;
-    private final WebSocketClientConnectorListener clientConnectorListener;
+    private final WebSocketClientListener clientConnectorListener;
     private final boolean readyOnConnect;
     private final ObjectValue webSocketClient;
-    private CountDownLatch countDownLatch;
 
     public WebSocketClientHandshakeListener(ObjectValue webSocketClient,
-                                            WebSocketService wsService,
-                                            WebSocketClientConnectorListener clientConnectorListener,
-                                            boolean readyOnConnect, CountDownLatch countDownLatch) {
+                                                WebSocketService wsService,
+                                                WebSocketClientListener clientConnectorListener,
+                                                boolean readyOnConnect) {
         this.webSocketClient = webSocketClient;
         this.wsService = wsService;
         this.clientConnectorListener = clientConnectorListener;
         this.readyOnConnect = readyOnConnect;
-        this.countDownLatch = countDownLatch;
     }
 
     @Override
     public void onSuccess(WebSocketConnection webSocketConnection, HttpCarbonResponse carbonResponse) {
         //using only one service endpoint in the client as there can be only one connection.
         webSocketClient.set(WebSocketConstants.CLIENT_RESPONSE_FIELD,
-                            HttpUtil.createResponseStruct(carbonResponse));
-        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(PROTOCOL_HTTP_PKG_ID,
-                                                                           WebSocketConstants.WEBSOCKET_CONNECTOR);
+                HttpUtil.createResponseStruct(carbonResponse));
+        ObjectValue webSocketConnector = BallerinaValues.createObjectValue(HttpConstants.PROTOCOL_HTTP_PKG_ID,
+                WebSocketConstants.WEBSOCKET_CONNECTOR);
         WebSocketOpenConnectionInfo connectionInfo = getWebSocketOpenConnectionInfo(webSocketConnection,
-                                                                                    webSocketConnector);
+                webSocketConnector);
         WebSocketUtil.populateWebSocketCaller(webSocketConnection, webSocketClient);
         clientConnectorListener.setConnectionInfo(connectionInfo);
         if (readyOnConnect) {
-            WebSocketUtil.readFirstFrame(webSocketConnection, webSocketClient);
+            webSocketConnection.readNextFrame();
         }
-        countDownLatch.countDown();
+        WebSocketUtil.countDownForHandshake(webSocketClient);
     }
 
     @Override
     public void onError(Throwable throwable, HttpCarbonResponse response) {
```

Comment:
Is the code in this function reused anywhere? If not why do we need an implementation here? We can declare it as abstract